src/rendering/vcTileRenderer.cpp

```diff
@@ -186,57 +220,103 @@ uint32_t vcTileRenderer_LoadThread(void *pThreadData)
         break;
       }
 
-      vcQuadTreeNode *pBestNode = pCache->tileLoadList[best];
-      pBestNode->renderInfo.loadStatus = vcNodeRenderInfo::vcTLS_Downloading;
-      pCache->tileLoadList.RemoveSwapLast(best);
-
-      char localFileName[vcMaxPathLength];
-      char serverAddress[vcMaxPathLength];
-
-      udSprintf(localFileName, "%s/%s/%d/%d/%d.%s", pRenderer->pSettings->cacheAssetPath, udUUID_GetAsString(pRenderer->pSettings->maptiles.tileServerAddressUUID), pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y, pRenderer->pSettings->maptiles.tileServerExtension);
-      udSprintf(serverAddress, "%s/%d/%d/%d.%s", pRenderer->pSettings->maptiles.tileServerAddress, pBestNode->slippyPosition.z, pBestNode->slippyPosition.x, pBestNode->slippyPosition.y, pRenderer->pSettings->maptiles.tileServerExtension);
-      udReleaseMutex(pCache->pMutex);
-
-      bool downloadingFromServer = true;
-      char *pTileURL = serverAddress;
-
-      if (udFileExists(localFileName) == udR_Success)
-      {
-        pTileURL = localFileName;
-        downloadingFromServer = false;
-      }
-
       udResult result = udR_Failure_;
+      udMutex *pMutexCopy = pCache->pMutex;
```

Comment:
I can see this function is much larger than I thought, but for future reference `udLockMutex` returns the pointer. So the pattern would be:
```c++
udMutex *pMutex = udLockMutex(pCache->pMutex);
// do some critical stuff
// some condition that might goto epilogue
// more critical stuff
udReleaseMutex(pMutex);
pMutex = nullptr;
// non critical work
epilogue:
udReleaseMutex(pMutex);
// clean up
return result;
```